pynemo/tide/nemo_bdy_tide3.py

```diff
@@ -1,272 +1,291 @@
+
+import logging
+
+def nemo_bdy_tpx7p2_rot(setup, DstCoord, Grid_T, Grid_U, Grid_V, comp):
+    """
+    TPXO Global Tidal model interpolation including rotation grid
+
+    INPUTS:
+        setup:                  settings
+        DstCoord:               ...
+        Grid_T         :        grid_type, bdy_r
+        Grid_U, Grid_V :        bdy_i , grid_type, bdy_r
+        comp:                   dictionary of harmonics read from namelist.
+                                e.g. {'1':"M2" , '2':"<constituent name>", ...}
+
+    RETURNS:
+        cosz, sinz, cosu, sinu, cosv, sinv: [# of constituents, number of bdy points]
+    """
+    key_transport = 0 # compute the velocities from transport
+    numharm = len(comp)
+    logger = logging.getLogger(__name__)
+    g_type = Grid_T.grid_type
+    DC = copy.deepcopy(DstCoord)
+    dst_lon = DC.bdy_lonlat[g_type]['lon'][Grid_T.bdy_r == 0]
+    dst_lat = DC.bdy_lonlat[g_type]['lat'][Grid_T.bdy_r == 0]
+
+    #nbdyz = len(Grid_T.bdy_i)
+    nbdyu = len(Grid_U.bdy_i)
+    nbdyv = len(Grid_V.bdy_i)
+
+    #convert the dst_lon into TMD Conventions (0E/360E)
+    dst_lon[dst_lon < 0.0] = dst_lon[dst_lon < 0.0]+360.0
+    #extract the surface elevation at each z-point
+    tpxo_z = tpxo_extract_HC.TpxoExtract(setup.settings, dst_lat, dst_lon, g_type)
+    #convert back the z-longitudes into the usual conventions (-180E/+180E)
+    dst_lon[dst_lon > 180.0] = dst_lon[dst_lon > 180.0]-360.0
+    #check if elevation data are missing
+    ind = np.where((np.isnan(tpxo_z.amp)) | (np.isnan(tpxo_z.gph)))
+    if ind[0].size > 0:
+        logger.warning('Missing elveation along the open boundary')
+
+    ampz = tpxo_z.amp
+    phaz = tpxo_z.gph
+    ampz[ind] = 0.0
+    phaz[ind] = 0.0
+
+    #extract U values of constituents
+    dst_lon = DC.bdy_lonlat[Grid_U.grid_type]['lon'][Grid_U.bdy_r == 0]
+    dst_lat = DC.bdy_lonlat[Grid_U.grid_type]['lat'][Grid_U.bdy_r == 0]
+    #set the array size for the target boundary output
+    if len(dst_lon) != len(dst_lon): logger.error('These should be the same size')
+    else: nbdyu = len(dst_lon)
+
+    #convert the U-longitudes into the TMD conventions (0/360E)
+    dst_lon[dst_lon < 0.0] = dst_lon[dst_lon < 0.0]+360.0
+
+    tpxo_ux = tpxo_extract_HC.TpxoExtract(setup.settings, dst_lat, dst_lon, Grid_U.grid_type)
+    tpxo_vx = tpxo_extract_HC.TpxoExtract(setup.settings, dst_lat, dst_lon, Grid_V.grid_type)
+
+    ampuX = tpxo_ux.amp
+    phauX = tpxo_ux.gph
+    ampvX = tpxo_vx.amp
+    phavX = tpxo_vx.gph
+
+    #check if ux data are missing
+    ind = np.where((np.isnan(ampuX)) | (np.isnan(phauX)))
+    if ind[0].size > 0:
+        logger.warning('Missing zonal velocity along the x open boundary')
+    ampuX[ind] = 0
+    phauX[ind] = 0
+    #check if vx data are missing
+    ind = np.where((np.isnan(ampvX)) | (np.isnan(phavX)))
+    if ind[0].size > 0:
+        logger.warning('Missing meridional velocity along the x open boundary')
```

Comment:
@jdha I've updated this warning message can you confirm this is the correct interpretation?